### ISSUE TITLE
Rubocop goes against Style/LambdaCalling

### ISSUE THREAD

**Issue by orotti13** [bbatsov/rubocop#4616]
Rubocop does not comply with ruby style guide when it comes to lambda calling.
https://github.com/bbatsov/ruby-style-guide#proc-call

```
# bad - looks similar to Enumeration access
l = ->(v) { puts v }
l[1]

# also bad - uncommon syntax
l = ->(v) { puts v }
l.(1)

# good
l = ->(v) { puts v }
l.call(1)
```

## Expected behavior

Rubocop shouldn't give warnings in these code samples:
`abcd.call`
`abcd.call(1)`
Whilst Rubocop is expected to do it here:
`abcd.()`
`abcd[]`
## Actual behavior

The actual behaviour is diametrically opposite from the expected. 
```
$ rubocop app/controllers/orders_controller.rb --only-guide-cops
<...>
app/controllers/orders_controller.rb:170:7: C: Prefer the use of lambda.(...) over lambda.call(...).
      pause.call
      ^^^^^^^^^^
```

## Steps to reproduce the problem

Simply copy-paste the provided samples.

## RuboCop version

```
0.49.1 (using Parser 2.4.0.0, running on ruby 2.3.3 x86_64-linux)
```

**Comment by rrosenblum** [comment]
@orotti13 I think this might be due to your RuboCop configuration. By default, `LambdaCall` is configured to use `call` which is what you want. 

```yml
Style/LambdaCall:
  EnforcedStyle: call 
  SupportedStyles:
    - call
    - braces
```
